### ISSUE TITLE
promisify .configure()

### ISSUE THREAD

**Issue by zapphyre** [martinj/node-mysql-promise#63]
please, can you make .configure() function return promise which would resolve when connected and reject with the reason when unable to connect?
it would be very handy for more reliable flow managment and testing too.

thx.

**Comment by martinj** [comment]
`.configure()` doesn't make any connections it only configures the mysql pool.
The pool will only try to open a connect when you make a query so it doesn't make sense to return a promise for `.configure()`.

I'm curious though how you think this would make a more reliable flow, can you provide any examples?

**Comment by zapphyre** [comment]
let me rephrase then: could you make  .configure() return a promise that would resolve upon usable configuration done and reject when there is any config error preventing dbs to work correctly? Maybe it can resolve an usable connection from the pool.
Ideally you would like to to know whether is the database in your program set up correctly before moving to initialization of other parts or before you do any queries which can fail.
Also for testing the good practice to keep things atomic, that is to test connection and then test querying/statements not to test connection by querying. 
My five cents..

**Comment by martinj** [comment]
I could expose a wrapper to `pool.getConnection` which you could use to validate connectivity to the database. Would that help?

**Comment by zapphyre** [comment]
yes, go ahead, please!

**Comment by martinj** [comment]
Don't forget you need to release the connection after retrieval. e.g https://github.com/martinj/node-mysql-promise/blob/5d4790160f5d03672244b68dd09f2e1e0360405d/test/index.test.js#L45-L55
